packages/block-library/src/post-author/edit.js

```diff
@@ -1,31 +1,256 @@
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+import classnames from 'classnames';
+
 /**
  * WordPress dependencies
  */
-import { useEntityProp, useEntityId } from '@wordpress/core-data';
+import apiFetch from '@wordpress/api-fetch';
+import { useRef, useState } from '@wordpress/element';
+import { useEntityProp } from '@wordpress/core-data';
+import {
+	AlignmentToolbar,
+	BlockControls,
+	FontSizePicker,
+	InspectorControls,
+	RichText,
+	__experimentalUseColors,
+	BlockColorsStyleSelector,
+	withFontSizes,
+} from '@wordpress/block-editor';
+import { PanelBody, SelectControl, ToggleControl } from '@wordpress/components';
 import { useSelect } from '@wordpress/data';
-import { sprintf, __ } from '@wordpress/i18n';
-
-function PostAuthorDisplay() {
-	const [ authorId ] = useEntityProp( 'postType', 'post', 'author' );
-	const author = useSelect(
-		( select ) =>
-			select( 'core' ).getEntityRecord( 'root', 'user', authorId ),
-		[ authorId ]
+import { __ } from '@wordpress/i18n';
+
+const DEFAULT_AVATAR_SIZE = 24;
+
+async function getAuthorDetails( authorId, setAuthorDetails ) {
+	const authorDetails = await apiFetch( {
+		path: '/wp/v2/users/' + authorId + '?context=edit',
+	} );
+	setAuthorDetails( authorDetails );
+}
```

Comment:
Maybe we can turn this into a selector and resolver pair in a follow-up PR :slightly_smiling_face: